packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts

```diff
@@ -1,28 +1,121 @@
 import inquirer from 'inquirer';
 import _ from 'lodash';
 import uuid from 'uuid';
+import path from 'path';
 import { LayerParameters, Permissions } from '../utils/layerParams';
 import { runtimeWalkthrough } from '../utils/functionPluginLoader';
+import { ServiceName, categoryName, layerParametersFileName } from '../utils/constants';
+import { objectExtension } from 'graphql-transformer-core/lib/TransformerContext';
 
 export async function createLayerWalkthrough(context: any, parameters: Partial<LayerParameters> = {}): Promise<Partial<LayerParameters>> {
   _.assign(parameters, await inquirer.prompt(layerNameQuestion(context)));
 
   let runtimeReturn = await runtimeWalkthrough(context, parameters);
   parameters.runtimes = runtimeReturn.map(val => val.runtime);
 
-  _.assign(parameters, await inquirer.prompt(layerPermissionsQuestion()));
+  _.assign(parameters, await inquirer.prompt(layerPermissionsQuestion(parameters)));
 
-  switch (parameters.layerPermissions) {
-    case Permissions.awsAccounts:
-      _.assign(parameters, await inquirer.prompt(layerAccountAccessQuestion()));
-      break;
-    case Permissions.awsOrg:
-      _.assign(parameters.authorizedOrgId, await inquirer.prompt(layerOrgAccessQuestion()));
-      break;
+  for (let permissions of parameters.layerPermissions) {
+    switch (permissions) {
+      case Permissions.awsAccounts:
+        _.assign(parameters, await inquirer.prompt(layerAccountAccessQuestion()));
+        break;
+      case Permissions.awsOrg:
+        _.assign(parameters, await inquirer.prompt(layerOrgAccessQuestion()));
+        break;
+    }
   }
   return parameters;
 }
 
+export async function updateLayerWalkthrough(
+  context: any,
+  templateParameters: Partial<LayerParameters> = {},
+): Promise<Partial<LayerParameters>> {
+  const { allResources } = await context.amplify.getResourceStatus();
+  const resources = allResources.filter(resource => resource.service === ServiceName.LambdaLayer).map(resource => resource.resourceName);
+
+  if (resources.length === 0) {
+    context.print.error('No Lambda Layer resource to update. Please use "amplify add function" command to create a new Function');
+    process.exit(0);
+    return;
+  }
+  const resourceQuestion = [
+    {
+      name: 'resourceName',
+      message: 'Please select the Lambda Layer you would want to update',
+      type: 'list',
+      choices: resources,
+    },
+  ];
+  const resourceAnswer = await inquirer.prompt(resourceQuestion);
+  templateParameters.layerName = String(resourceAnswer.resourceName);
```

Comment:
What's the purpose of this type conversion? When is `resourceAnswer.resourceName` not a string?

Reply:
Ya , I didnt checked that earlier , thanks
fixed